### PR TITLE
Router implementation

### PR DESCRIPTION
**Pull request by VVRud**
Added basic router implementation.

### CODE REVIEW

File: router/database/crud.go [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package database
+
+import (
+	"database/sql"
+	"serverless/router/schema"
+)
+
+func SaveUser(db *sql.DB, creds schema.Credentials) error {
+	_, err := db.Exec("INSERT INTO users (username, password) VALUES ($1, $2)", creds.Username, creds.Password)
```
Comment (maxpushka) [inline router/database/crud.go:9]: I suggest modifying the code to hash the password before saving it to the database. For example, you can use a library like `golang.org/x/crypto/bcrypt` to hash the password.

The password in auth handlers should be compared against hash as well

File: config/config.go [hunk 2]
```diff
@@ -1,8 +1,53 @@
...
+	AuthJWTKey     []byte
+	AuthJWTExpires time.Duration
+	// Server Config
+	ServerPort string
+	// DB Config
+	dbHost string
+	dbPort string
+	dbUser string
+	dbPass string
+	dbName string
```
Comment (maxpushka) [inline config/config.go:21]: I suggest moving the DB-related fields into a `Database` struct making config more modular. The `DbUrl` method below could be associated with it as well.

File: router/router.go [hunk 3]
```diff
@@ -1 +1,55 @@
...
+	//"database/sql"
+	//"encoding/json"
+	"fmt"
+	//"io"
+	"log"
+	"net/http"
+	//"os"
+	"serverless/router/database"
+	//"strings"
+	//"time"
```
Comment (maxpushka) [inline router/router.go:14]: Commented-out imports add unnecessary clutter to the code and do not provide any functional or informational value. I suggest removing them to improve code readability and maintainability

File: config/config.go [hunk 4]
```diff
@@ -1,8 +1,53 @@
...
+}
+
+func New() (*Config, error) {
+	err := godotenv.Load()
+	if err != nil {
+		log.Fatal("Error loading .env file")
+		return nil, err
+	}
```
Comment (maxpushka) [inline config/config.go:29]: A non-blocking comment for discussion: could we benefit from moving config into a YAML or TOML format? I believe there should be a lot of config entries added once CDN component is finished

File: router/router.go [hunk 5]
```diff
@@ -1 +1,55 @@
...
+	}).Methods("POST")
+	r.HandleFunc("/register", func(w http.ResponseWriter, r *http.Request) {
+		auth.Register(db, conf, w, r)
+	}).Methods("POST")
+
+	api := r.PathPrefix("/api").Subrouter()
+	api.Use(auth.Middleware(conf))
+
+	fmt.Printf("Starting server on port %s\n", conf.ServerPort)
+	err = http.ListenAndServe(":"+conf.ServerPort, r)
```
Comment (maxpushka) [inline router/router.go:47]: The current impl implies listening on loopback interface (`127.0.0.1` for IPv4), which restricts access to the local machine only. To allow connections from all network interfaces, let's update the code to explicitly bind to `0.0.0.0:<port>`. This ensures the server is accessible externally, which would be useful once we come around to deployments in the cloud.

**Review by maxpushka — CHANGES_REQUESTED** [review]

File: config/config.go (hunk 4 above)
Comment (VVRud) [reply config/config.go:29]: Yep, I think that can be a good idea, actually

**Pull request merged**